File: C++/2661. First Completely Painted Row or Column.cpp
class Solution {
public:
    int firstCompleteIndex(vector<int>& a, vector<vector<int>>& m) {
        map<int,int>r,c;
        for(int i=0;i<m.size();i++){
            for(int j=0;j<m[0].size();j++){
                r[m[i][j]]=i;
                c[m[i][j]]=j;
            }
        }
        vector<int>x(m.size(),0),y(m[0].size(),0);
        for(int i=0;i<a.size();i++){
            if(++x[r[a[i]]]==m[0].size())
            return i;
            if(++y[c[a[i]]]==m.size())
            return i;
        }
        return -1;
    }
};

File: JAVA/2661. First Completely Painted Row or Column.java
class Solution {
    public int firstCompleteIndex(int[] a, int[][] m) {
        int[] r=new int[m.length*m[0].length+1];
        int[] c=new int[m.length*m[0].length+1];
        int[] x=new int[m.length];
        int[] y=new int[m[0].length];
        for(int i=0;i<m.length;i++){
            for(int j=0;j<m[0].length;j++){
                r[m[i][j]]=i;
                c[m[i][j]]=j;
            }
        }
        for(int i=0;i<a.length;i++){
            if(++x[r[a[i]]]==m[0].length)
            return i;
            if(++y[c[a[i]]]==m.length)
            return i;
        }
        return -1;
    }
}